File: 프로그래머스/unrated/159994. 카드 뭉치/카드 뭉치.py
def solution(cards1, cards2, goal):
    answer="Yes"
    for g in goal:
        if cards1 and g==cards1[0] :
            cards1.remove(g)
        elif cards2 and g==cards2[0]:
            cards2.remove(g)
        else :
            answer="No"
        
    return answer

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
import java.util.*;
class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        
        List<String> c1=new ArrayList<>(Arrays.asList(cards1));
        List<String> c2=new ArrayList<>(Arrays.asList(cards2));
        
        String answer = "Yes";
        for(int i=0;i<goal.length;i++){
            if(!c1.isEmpty()&&c1.get(0).equals(goal[i])) c1.remove(0);
            else if(!c2.isEmpty()&&c2.get(0).equals(goal[i]))c2.remove(0);
            else{
                answer="No";
                break;
            }
        }
        return answer;
    } 
}